subsys/net/ip/icmpv4.c

```diff
@@ -59,13 +67,415 @@ int net_icmpv4_finalize(struct net_pkt *pkt)
 	return net_pkt_set_data(pkt, &icmpv4_access);
 }
 
+/* Parse Record Route and add our own IP address based on
+ * free entries.
```

Comment:
Hmm, I'd definitely want to see any IP option parsing centrally in one place, so that the IP options supported can be handled once in a central location. It is a good idea to update IPv4 header length check functions so that they always consider option lengths (but that may have been done already anyway?)

Reply:
I am all for parsing IP options in one place. It is a bit challenging here as information (basically the packet data) about the parsed options would need to be passed somehow to icmpv4 code which would then need to send the packet according to that information. The information here can be max. 40 bytes long.

Reply:
I would actually push just the option parsing code into something else than icmp files. I would also skip the option parsing in the ipv4 code by default and call the option parsing code from exactly here in icmp and/or in similar places, once there are options that are supposed to be handled by a certain use case. Those use cases should then enable Kconfig variables to make this happen as not all Zephyr use cases need to care about options. Since the packets containing these options usually seem to fall off when filtering, maybe the test tool will get its own set of option to pass one of its tests? That means there still are tests for option parsing and a guarantee that the code works. Thus we ensure that other use cases are not affected by extra option checks if they don't need to care about any options?

So for this code it just means to push option parsing into a file of its own and set a relevant Kconfig variable in order to check options here. That would not be a too big modification?

Reply:
Ok, whatever. Unusal case anyway.